ext/session/session.c

```diff
@@ -377,6 +378,7 @@ PHPAPI zend_result php_session_valid_key(const char *key) /* {{{ */
 	/* Somewhat arbitrary length limit here, but should be way more than
 	   anyone needs and avoids file-level warnings later on if we exceed MAX_PATH */
 	if (len == 0 || len > PS_MAX_SID_LENGTH) {
+		php_error_docref(NULL, E_WARNING, "Prefix or name cannot be larger than 256 characters");
```

Comment:
Maybe `Prefix or name cannot be empty or larger than 256 characters` what do you think ?

Reply:
I'm not sure if we should raise warnings from `php_session_valid_key()`at all, since that may be used by external session handlers which may already raise sensible warnings on FAILURE.

Reply:
I think I agree, it might be better to create an enum and return this instead of `zend_result` with the different failure cases, so that the warning/exceptions at call site can be more appropriate.

Reply:
That's a valid point. Do you think we can just add a warning about prefix size to the existing error message?

Reply:
Yes this is fine